pkg/modules/observability.go

```diff
@@ -0,0 +1,141 @@
+package modules
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	csmv1 "github.com/dell/csm-operator/api/v1alpha1"
+	drivers "github.com/dell/csm-operator/pkg/drivers"
+	"github.com/dell/csm-operator/pkg/logger"
+	utils "github.com/dell/csm-operator/pkg/utils"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+
+	// ObservabilityTopologyName - component topology
+	ObservabilityTopologyName string = "topology"
+
+	// ObservabilityMetricsPowerScaleName - component metrics-powerscale
+	ObservabilityMetricsPowerScaleName string = "metrics-powerscale"
+
+	// ObservabilityMetricsPowerFlexName - component metrics-powerflex
+	ObservabilityMetricsPowerFlexName string = "metrics-powerflex"
+
+	// TopologyLogLevel -
+	TopologyLogLevel string = "<TOPOLOGY_LOG_LEVEL>"
+
+	// TopologyImage -
+	TopologyImage string = "<TOPOLOGY_IMAGE>"
+
+	// TopologyYamlFile -
+	TopologyYamlFile string = "karavi-topology.yaml"
+)
+
+// ObservabilitySupportedDrivers is a map containing the CSI Drivers supported by CMS Replication. The key is driver name and the value is the driver plugin identifier
+var ObservabilitySupportedDrivers = map[string]SupportedDriverParam{
+	"powerscale": {
+		PluginIdentifier:              drivers.PowerScalePluginIdentifier,
+		DriverConfigParamsVolumeMount: drivers.PowerScaleConfigParamsVolumeMount,
+	},
+	"isilon": {
+		PluginIdentifier:              drivers.PowerScalePluginIdentifier,
+		DriverConfigParamsVolumeMount: drivers.PowerScaleConfigParamsVolumeMount,
+	},
+}
+
+// ObservabilityPrecheck  - runs precheck for CSM Observability
+func ObservabilityPrecheck(ctx context.Context, op utils.OperatorConfig, topo csmv1.Module, cr csmv1.ContainerStorageModule, r utils.ReconcileCSM) error {
```

Comment:
better rename `topo csmv1.Module` to `obs csmv1.Module`

Reply:
addressed